File: algos/ugly_number/ts_ugly_number/src/ans.ts
function isUgly(n: number): boolean {
  if (n === 0) { return false; }
  let divisors = [2,3,5];
  divisors.forEach((x)=>{while (n%x==0) {n /= x}});
  return n === 1;
};

export {isUgly};

File: algos/ugly_number/cpp_ugly_number/src/ans.cpp
#include "ans.hpp"

bool Solution::isUgly(int n) {
  if (n == 0) { return false; }
  int divisors[3] = {2,3,5};
  for (auto divisor : divisors) {
    while (n % divisor == 0) { n /= divisor; }
  }
  return n == 1;
}

